python/lsst/ip/isr/isrFunctions.py

```diff
@@ -592,6 +594,230 @@ def brighterFatterCorrection(exposure, kernel, maxIter, threshold, applyGain, ga
+    counts and the correction.  We stop iterating when the summed
+    difference between the current corrected image and the one from
+    the previous iteration is below the threshold.  We do not require
+    convergence because the number of iterations is too large a
+    computational cost.  How we define the threshold still needs to be
+    evaluated, the current default was shown to work reasonably well
+    on a small set of images.
+    Edges are handled in the convolution by padding.  This is still not
+    a physical model for the edge, but avoids discontinuity in the correction.
+
+    Author of modified version: Lance.Miller@physics.ox.ac.uk
+    (see DM-33555).
+    """
+
+    image = exposure.getMaskedImage().getImage()
+
+    # The image needs to be units of electrons/holes
+    with gainContext(exposure, image, applyGain, gains):
+
+        # get kernel and its shape
+        kLy, kLx = kernel.shape
+        kernelImage = afwImage.ImageD(kLx, kLy)
+        kernelImage.getArray()[:, :] = kernel
+        tempImage = image.clone()
+
+        nanIndex = numpy.isnan(tempImage.getArray())
+        tempImage.getArray()[nanIndex] = 0.
+
+        outImage = afwImage.ImageF(image.getDimensions())
+        corr = numpy.zeros_like(image.getArray())
+        prevImage = numpy.zeros_like(image.getArray())
+        convCntrl = afwMath.ConvolutionControl(False, True, 1)
+        fixedKernel = afwMath.FixedKernel(kernelImage)
+
+        # set the padding amount
+        # ensure we pad by an even amount larger than the kernel
+        kLy = 2 * ((1+kLy)//2)
+        kLx = 2 * ((1+kLx)//2)
+
+        # The deflection potential only depends on the gradient of
+        # the convolution, so we can subtract the mean, which then
+        # allows us to pad the image with zeros and avoid wrap-around effects
+        # (although still not handling the image edges with a physical model)
+        # This wouldn't be great if there were a strong image gradient.
+        imydim, imxdim = tempImage.array.shape
+        imean = numpy.mean(tempImage.getArray()[~nanIndex])
+        # subtract mean from image
+        tempImage -= imean
+        tempImage.array[nanIndex] = 0.
+        padArray = numpy.pad(tempImage.getArray(), ((0, kLy), (0, kLx)))
+        outImage = afwImage.ImageF(numpy.pad(outImage.getArray(), ((0, kLy), (0, kLx))))
+        # Convert array to afw image so afwMath.convolve works
+        padImage = afwImage.ImageF(padArray.shape[1], padArray.shape[0])
+        padImage.array[:] = padArray
+
+        for iteration in range(maxIter):
+
+            # create deflection potential, convolution of flux with kernel
+            # using padded counts array
+            afwMath.convolve(outImage, padImage, fixedKernel, convCntrl)
+            tmpArray = tempImage.getArray()
+            outArray = outImage.getArray()
+
+            # trim convolution output back to original shape
+            outArray = outArray[:imydim, :imxdim]
```

Comment:
This needs to match above as well (I think this is the only place they're used).